File: enode/src/main/java/org/enodeframework/common/io/Task.java
package org.enodeframework.common.io;

import io.vertx.ext.sync.Sync;
import org.enodeframework.common.exception.ENodeInterruptException;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CountDownLatch;

/**
 * @author anruence@gmail.com
 */
public class Task {

    public static CompletableFuture<Void> completedTask = CompletableFuture.completedFuture(null);

    public static void await(CountDownLatch latch) {
        try {
            latch.await();
        } catch (InterruptedException e) {
            throw new ENodeInterruptException(e);
        }
    }

    public static <T> T await(CompletableFuture<T> future) {
        return future.join();
    }

    public static void sleep(long sleepMilliseconds) {
        try {
            Thread.sleep(sleepMilliseconds);
        } catch (InterruptedException e) {
            throw new ENodeInterruptException(e);
        }
    }
}

File: enode/src/main/java/org/enodeframework/common/io/Task.kt
package org.enodeframework.common.io

import org.enodeframework.common.exception.EnodeInterruptException
import java.util.concurrent.CompletableFuture
import java.util.concurrent.CountDownLatch

/**
 * @author anruence@gmail.com
 */
object Task {
    @JvmField
    var completedTask: CompletableFuture<Boolean> = CompletableFuture.completedFuture(true)

    @JvmStatic
    fun await(latch: CountDownLatch) {
        try {
            latch.await()
        } catch (e: InterruptedException) {
            throw EnodeInterruptException(e)
        }
    }

    @JvmStatic
    fun <T> await(future: CompletableFuture<T>): T {
        return future.join()
    }

    @JvmStatic
    fun sleep(sleepMilliseconds: Long) {
        try {
            Thread.sleep(sleepMilliseconds)
        } catch (e: InterruptedException) {
            throw EnodeInterruptException(e)
        }
    }
}